include/clap/ext/draft/preset-load.h

```diff
@@ -9,19 +9,13 @@ extern "C" {
 #endif
 
 typedef struct clap_plugin_preset_load {
-   // Loads a preset in the plugin native preset file format from a path.
-   // [main-thread]
-   bool(CLAP_ABI *from_file)(const clap_plugin_t *plugin, const char *path);
-
-   // Loads a preset in the plugin native preset file format from a path which points to a preset
-   // container file.
-   // preset_id must be a valid string, which is specific to the plugin itself and identifies the
-   // desired preset within the given preset container.
+   // Loads a preset in the plugin native preset file format from a URI. eg:
+   // - "file:///home/abique/.u-he/Diva/Presets/Diva/HS Bass Nine.h2p"
+   // - "file:///home/abique/my-sound-bank/<preset-id>"
```

Comment:
How about using a colon for targetting presets inside of bank files? Kinda like how you specify a path on a remote machine (`user@hostname:path/to/file`). Would that remove a layer of ambiguity (you now need to first check if `/home/abique/my-sound-bank` is a file or directory) or would that only make things more complex? Then the example would become `file:///home/abique/my-sound-bank:<preset-id>`.

Reply:
I think it becomes then more difficult to parse isn't it?
Because `:` is a valid filename character on Linux while `/` isn't.
I believe that using `/` makes it easier to break the path using standard path manipulation libraries.
What do you think?

Reply:
You could ASCII 0x1F Unit Separator. That's unlikely to conflict with any real file paths.

Reply:
It should be easier to parse, which is the motivation behind the suggestion. `:` is a valid character in file paths, but the same notation has been used a lot for referencing specific lines in files. I just checked what grep and co do when file names contain colons:

```
/tmp/grep-party
❯ lt
drwxr-xr-x   - robbert  3 jan 18:37 .
drwxr-xr-x   - robbert  3 jan 18:38 ├── dir-containing-a-:-character
.rw-r--r-- 292 robbert  3 jan 18:38 │  ├── also-containing-a-:-character
.rw-r--r-- 292 robbert  3 jan 18:38 │  ├── some-file
.rw-r--r-- 292 robbert  3 jan 18:38 │  └── trailing-:
.rw-r--r-- 292 robbert  3 jan 18:37 └── file-containing-a-:-character

/tmp/grep-party
❯ grep -R 99
dir-containing-a-:-character/trailing-::99
dir-containing-a-:-character/also-containing-a-:-character:99
dir-containing-a-:-character/some-file:99
file-containing-a-:-character:99

/tmp/grep-party
❯ rg 99
file-containing-a-:-character
99:99

dir-containing-a-:-character/some-file
99:99

dir-containing-a-:-character/trailing-:
99:99

dir-containing-a-:-character/also-containing-a-:-character
99:99

/tmp/grep-party
❯ rg --vimgrep 99
file-containing-a-:-character:99:1:99
dir-containing-a-:-character/some-file:99:1:99
dir-containing-a-:-character/also-containing-a-:-character:99:1:99
dir-containing-a-:-character/trailing-::99:1:99
```

So they don't treat colons differently, and the last colon is the separator. That means that preset IDs cannot contain the separator. So regardless of what separator is used for banks that also needs to be documented. Or if slashes _are_ allowed in preset names, then that should also be documented. So a `bass/gnarly-reese` preset in the `user.dat` bank would be addressed as `file:///home/username/.local/share/awesome-synth/presets/user.dat/bass/gnarly-reese`. In that case the path just always needs to be parsed segment by segment.

I guess I don't have a strong opinion here. Colons might make parsing easier and it at least makes it easier to distinguish bank file paths from preset IDs for humans, but the URI parsing needs to be implemented with care either way.

Reply:
One other option is to add a `set_load_uri()` in the metadata receiver, then the plugin is responsible for defining the load uri.

Reply:
Otherwise yes using `/` forces you to stat the parent until it is a file, but it is OK and only necessary for container presets.